### DISCUSSION TITLE
about upx_dir,I can't use cmd --upx-dir to set my upx path effective

### DISCUSSION THREAD

**Discussion by MRWER** [pyinstaller/pyinstaller#8548] [Help] [answered]
![image](https://github.com/pyinstaller/pyinstaller/assets/7991237/cf4ef7a8-d655-460b-b581-392c08b3ae7d)
when i use --upx-dir="E:\my_project\upx.exe" to build, it wouldn't recognition correct，because the "upx_dir" has value and then the "upx_exe" would be "E:\my_project\upx.exe\upx", so "subprocess.check_output" went wrong. but if --upx-dir="E:\my_project\" the "upx_exe" would be "E:\my_project\upx", it also don't work.
 Can someone tell me the correct way to use it？ /(ㄒoㄒ)/~~

**Comment by rokm** [comment, marked as answer]
> but if --upx-dir="E:\my_project" the "upx_exe" would be "E:\my_project\upx", it also don't work.

That should work. And it does for me.

What happens if you open python shell and try to manually run
``` 
import subprocess
subprocess.check_output([r'E:\my_project\upx', '-V'], stdin=subprocess.DEVNULL, stderr=subprocess.DEVNULL, encoding='utf8')
```

**Reply by rokm** [reply]
But instead of using `--upx-dir`, you could also ensure that upx executable is in `PATH` prior to running pyinstaller:

``` 
set PATH=E:\my_project;%PATH%
pyinstaller ...
```

**Reply by MRWER** [reply]
thanks for help~ when i restart my VsCode, it done well......
set PATH is ok, I  just want to test the cmd, lol~
